### PR TITLE
feat(client): ts-migration for Introduction templates

### PR DESCRIPTION
**Pull request by Ismailtlem**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Related to the issue  #43778

### CODE REVIEW

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Thank you, for taking the time to contribute again, Ismail.

This PR is solid, but there is a glaring issue that TS annoyingly does not recognise, because Redux is not typed yet.

Otherwise, could you also rename the files to `kebab-case`?

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Sorry about this again, but I got confused with the typing, because I missed the transformation in the dispatch.

**Review by ShaunSHamilton — DISMISSED** [review]
Thank you, for bearing with me. I resolved the conflicts, and pushed minor changes.

TLGTM 👍

File: client/src/redux/prop-types.ts [hunk 1]
```diff
@@ -69,6 +70,14 @@ export const StepsPropType = PropTypes.shape({
...
 });
 
+export type Steps = {
+  isHonest?: boolean;
+  currentCerts?: Array<CurrentCert>;
+  isShowCerts?: boolean;
+  isShowName?: boolean;
+  isShowProfile?: boolean;
+};
+
```
Comment (ojeytonwilliams) [inline client/src/redux/prop-types.ts:80]: Now we have this, can we remove the StepsPropType?

**Review by BIOSTYMA — APPROVED** [review]

**Review by moT01 — DISMISSED** [review]
LGTM @Ismailtlem :tada:

**Review by ShaunSHamilton — APPROVED** [review]
I just added some types to remove the need for `eslint-disable` lines.

@moT01 Please merge, if you are still happy.

**Review by moT01 — APPROVED** [review]
LGTM @Ismailtlem :tada: Thanks for contributing to freeCodeCamp :tada:

**Pull request merged**